### DISCUSSION TITLE
Manually/Live updating the list

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#80] [Ideas]
https://user-images.githubusercontent.com/36073/112730162-a6b5dc00-8ef5-11eb-8df5-b376a0cae44f.mp4

Note: This has no impact on existing scripts/APIs. Just adding more API/control. These scripts would not work in the terminal (for now), but I'm moving forward with adding UI features and then finding ways of bringing them back to the terminal.

I'm pretty sold on this idea. I'm definitely moving forward with internally separating updating the prompt from the choices (these used to be in the same initial call). Curious if anyone can think of use-cases that I might need to cover.

**Comment by zzeleznick** [comment]
I was playing around with RxJs, which is how I found this project (via author [mention](https://www.learnrxjs.io/learn-rxjs/recipes/progressbar)), and would like to build a countdown timer in Kit. 

Would it be possible to keep Kit application window alive until the timer (i.e. rendered via a dynamic choice) is stopped or completes?

**Reply by johnlindquist** [reply]
I added a Roadmap discussion for [Script Widgets](https://github.com/johnlindquist/kit/discussions/82) which would be perfect for timers, etc. I've been thinking about this for a loooooong time and I'm excited to bring it all together.

**Reply by zzeleznick** [reply]
Amazing, thanks for all the work here behind the scenes and kicking off the the Roadmap Discussion!

Riffing on the demo you shared above, I played around with some of the internals to show the current unix time:

## Demo
![kit-time](https://user-images.githubusercontent.com/5779832/112763810-e5619a00-8fba-11eb-9dbe-9e548981ce74.gif)

<details>
<summary>
Click for code
</summary>

```js
// Menu: Kit Time
// Description: Display time
// Author: Zach Zeleznick
// Twitter: @zzxiv

// inspired by https://time.is/Unix

let interval;

const now = () => (new Date()).getTime()

const buildDisplay = () => {
  const t = now();
  return {
    name: t.toString().slice(0,10),
    value: t,
    description: Date(t),
  }
}

const updateChoices = () => {
  process.send({from: 'UPDATE_PROMPT_CHOICES', choices: [ buildDisplay() ]});
}

interval = setInterval(updateChoices, 1000);

await arg("Unix time now", [ buildDisplay() ]);

```
</details>

**Comment by jsjoeio** [comment]
It's a cool idea, but no immediate use-cases are coming to mind. Hard to imagine what I might use this for myself.
